crates/runner-host/crates/web-client/src/board.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use log::info;
+use web_common::{Command, Component, Event};
+use yew::prelude::*;
+use yew::{function_component, html, Html, Properties};
+use yew_hooks::use_set;
+
+use crate::board_components::button::Button;
+use crate::board_components::led::LED;
+
+#[derive(Properties, PartialEq)]
+pub struct Props {
+    pub command_state: UseStateHandle<Option<Command>>,
+    pub on_board_ready: Callback<()>,
+    pub on_event: Callback<Event>,
+}
+
+#[function_component]
+pub fn Board(Props { command_state, on_board_ready, on_event }: &Props) -> Html {
+    let board_config = use_state_eq(|| None);
+    let enabled_leds = use_set(HashSet::new());
```

Comment:
This looks weird to me that Yew forces the state of sub-components to be in the parent. Either there is a better way using function components, or we have to use struct components in our case. The `Component` interface looks more appropriate to message passing to me (using the `update` function and thus matching the messages we get on the websocket), than those hooks.

Did you try implementing `Component` instead of using `function_component` macro? I would expect a design where the app would start a websocket and its update function would be called each time there is a message (not sure how that would be done), and then the app would dispatch the message to the appropriate sub-component.

Reply:
I reworked this to store the state of the led inside of the led component and just pass it the command state so it can update itself. I think it should be possible to do that with arbitrary function components with a bit of thinking so I think we should be able to avoid stuct components. 

Let me know what you think of the new version. If that is the type of thing you were going for.

Reply:
Yes, this looks good. Thanks!